tests/message_processing/test_process_instances.py

```diff
@@ -518,3 +626,33 @@ async def test_compare_cost_view_with_cost_function(
         ).scalar_one()
 
     assert Decimal(str(cost_from_view)) == cost_from_function
+
+
+@pytest.mark.asyncio
+async def test_compare_cost_view_with_cost_function_payg(
+    session_factory: DbSessionFactory,
+    message_processor: PendingMessageProcessor,
+    fixture_instance_message_payg: PendingMessageDb,
+    user_balance: AlephBalanceDb,
+):
+    with session_factory() as session:
+        insert_volume_refs(session, fixture_instance_message_payg)
+        session.commit()
+
+    pipeline = message_processor.make_pipeline()
+    # Exhaust the iterator
+    _ = [message async for message in pipeline]
+
+    assert fixture_instance_message_payg.item_content
+    content = InstanceContent.parse_raw(
+        fixture_instance_message_payg.item_content
+    )  # Parse again
+    with session_factory() as session:
+        assert content.payment.type == PaymentType.superfluid
+        cost_from_view = session.execute(
+            text("SELECT total_price from vm_costs_view WHERE vm_hash = :vm_hash"),
```

Comment:
Shouldn't we use sqlalchemy directly here for such a small query?

Or is it because the view is only done in full sql?

Reply:
Yeah it's pure sql on the view we only have accessor for the cost_view  
based on the address but I could add one to avoid code duplications also could be useful for future feature 

(exemple having more details or the locked amount)